### DISCUSSION TITLE
Query.update() for joined table

### DISCUSSION THREAD

**Discussion by liverpool1026** [sqlalchemy/sqlalchemy#10128] [Usage Questions] [answered]
It's been a while, but we just recently migrated from 1.3 -> 1.4. And I have some question regarding Query.update() for joined table.

So we have the following table structure

```
class Base:
     ....

class Child(Base):
    .....
```

And in 1.3 we did this

```
data_set.query(Child).update({Child.column: some_value}, synchronize_session="fetch")
```
and it works. 
Looking at the 1.3 docs https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query.update 
It did mentioned that **However, the join condition of an inheritance mapper is not automatically rendered.**
Which obviously nothing is handled here, so I am wondering if this is just working in some weird way?

As in 1.4, the same code now is throwing an error
![image](https://github.com/sqlalchemy/sqlalchemy/assets/29122286/bd953c40-983e-45f1-81cd-98d5495c4a4c)

![image](https://github.com/sqlalchemy/sqlalchemy/assets/29122286/e291e872-2daa-4989-aa8d-96590c351d67)
Which in the docs, it still mentioned that Joined Inheritance is not supported.

Basically, my question is why does this work in 1.3? Is it due to some reason that I am not aware of?

**Comment by zzzeek** [comment, marked as answer]
it would be undefined behavior of some kind, and possibly even "works" by updating more rows than you'd expect, so I'd be careful with that, look at the SQL logs and SQL emitted to ensure it's what you intended.

**Reply by liverpool1026** [reply]
Thought so, thanks
